tdrs-backend/tdpservice/parsers/test/test_parse.py

```diff
@@ -1555,6 +1550,137 @@ def test_parse_tribal_section_4_file(tribal_section_4_file, dfs):
+        file__section=DataFile.Section.ACTIVE_CASE_DATA,
+        file__data=(b'HEADER20211A25   TAN1ED\n' +
+                    b'T320210211111111157120190527WTTTTT9WT12212122204398100000000' +
+                    b'420100125WTTTT9@TB1221222220430490000\n' +
+                    b'TRAILER0000001         ')
+    )
+    return parsing_file
+
+@pytest.fixture
+def t3_file_two_child_with_space_filled():
+    """Fixture for T3 file."""
+    # T3 record is space filled correctly
+    parsing_file = ParsingFileFactory(
+        year=2021,
+        quarter='Q2',
+        original_filename='t3_file_two_child_with_space_filled.txt',
+        file__name='t3_file_two_child_with_space_filled.txt',
+        file__section=DataFile.Section.ACTIVE_CASE_DATA,
+        file__data=(b'HEADER20211A25   TAN1ED\n' +
+                    b'T320210211111111157120190527WTTTTT9WT12212122204398100000000' +
+                    b'420100125WTTTT9@TB1221222220430490000                       \n' +
+                    b'TRAILER0000001         ')
+    )
+    return parsing_file
+
+
+@pytest.fixture
+def two_child_second_filled():
+    """Fixture for T3 file."""
+    # T3 record is space filled correctly
+    parsing_file = ParsingFileFactory(
+        year=2021,
+        quarter='Q2',
+        original_filename='two_child_second_filled.txt',
+        file__name='two_child_second_filled.txt',
+        file__section=DataFile.Section.ACTIVE_CASE_DATA,
+        file__data=(b'HEADER20211A25   TAN1ED\n' +
+                    b'T320210211111111115120160401WTTTT@BTB22212212204398100000000' +
+                    b'56      111111111                                           ' +
+                    b'                                    \n' +
+                    b'TRAILER0000001         ')
+    )
+    return parsing_file
+
+
+@pytest.mark.parametrize('file_fixture, result, number_of_errors, error_message',
+                         [('second_child_only_space_t3_file', True, 0, ''),
+                          ('one_child_t3_file', True, 0, ''),
+                          ('t3_file', True, 0, ''),
+                          ('t3_file_two_child', True, 1,
+                           'The second child record is too short at 97 characters' +
+                           ' and must be at least 101 characters.'),
+                          ('t3_file_two_child_with_space_filled', True, 0, ''),
+                          ('two_child_second_filled', True, 9, 'T3: Year 6    must be larger than 1900.')])
+@pytest.mark.django_db()
+def test_misformatted_multi_records(file_fixture, result, number_of_errors, error_message, request, dfs):
+    """Test that (not space filled) multi-records are caught."""
+    file_fixture = request.getfixturevalue(file_fixture)
+    dfs.datafile = file_fixture
+    parse.parse_datafile(file_fixture, dfs)
+    parser_errors = ParserError.objects.filter(file=file_fixture)
+    t3 = TANF_T3.objects.all()
+    assert t3.exists() == result
```

Comment:
another suggestion is to assert on the number of records created (1 or 2 depending on the case)